pallets/api/src/nonfungibles/mod.rs

```diff
@@ -0,0 +1,664 @@
+		pub fn burn(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: ItemIdOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let owner = NftsOf::<T>::owner(collection, item)
+				.ok_or(NftsErrorOf::<T>::UnknownItem)
+				.map_err(|e| e.with_weight(T::DbWeight::get().reads(1)))?;
+			NftsOf::<T>::burn(origin, collection, item)?;
+			Self::deposit_event(Event::Transfer { collection, item, from: Some(owner), to: None });
+			Ok(().into())
+		}
+
+		/// Issue a new collection of non-fungible items.
+		///
+		/// # Parameters
+		/// - `admin` - The admin of this collection.
+		/// - `config` - The configuration of the collection.
+		#[pallet::call_index(12)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::create() + T::DbWeight::get().reads(1))]
+		pub fn create(
+			origin: OriginFor<T>,
+			admin: AccountIdOf<T>,
+			config: CollectionConfigOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let creator = ensure_signed(origin.clone())?;
+			let id = NextCollectionIdOf::<T>::get()
+				.or(T::CollectionId::initial_value())
+				.ok_or(NftsErrorOf::<T>::UnknownCollection)
+				.map_err(|e| e.with_weight(T::DbWeight::get().reads(1)))?;
+			NftsOf::<T>::create(origin, T::Lookup::unlookup(admin.clone()), config)?;
+			Self::deposit_event(Event::Created { id, creator, admin });
+			Ok(().into())
+		}
+
+		/// Destroy a collection of items.
+		///
+		/// # Parameters
+		/// - `collection` - The collection to destroy.
+		/// - `witness` - Information on the items minted in the `collection`. This must be
+		/// correct.
+		#[pallet::call_index(13)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::destroy(
+    		witness.item_metadatas,
+    		witness.item_configs,
+    		witness.attributes,
+		))]
+		pub fn destroy(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			witness: DestroyWitness,
+		) -> DispatchResultWithPostInfo {
+			NftsOf::<T>::destroy(origin, collection, witness)
+		}
+
+		/// Set an attribute for a collection or item.
+		///
+		/// # Parameters
+		/// - `collection` - The collection identifier.
+		/// - `item` - Optional item whose attribute to set. If not provided, the `collection`'s
+		///   attribute is set.
+		/// - `namespace` - Attribute's namespace.
+		/// - `key` - The key of the attribute.
+		/// - `value` - The value to which to set the attribute.
+		#[pallet::call_index(14)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::set_attribute())]
+		pub fn set_attribute(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: Option<ItemIdOf<T>>,
+			namespace: AttributeNamespaceOf<T>,
+			key: BoundedVec<u8, T::KeyLimit>,
+			value: BoundedVec<u8, T::ValueLimit>,
+		) -> DispatchResult {
+			NftsOf::<T>::set_attribute(
+				origin,
+				collection,
+				item,
+				namespace,
+				key.clone(),
+				value.clone(),
+			)?;
+			Self::deposit_event(Event::AttributeSet {
+				collection,
+				item,
+				key: key.to_vec(),
+				data: value.to_vec(),
```

Comment:
We are currently cloning each of these twice. Once when you do `key.clone()`, and once when you do `key.to_vec()`

I recommend defining two vars at the top of the fn: `key_vec = key.to_vec();` and `data_vec = data.to_vec()`. This then allows you to do only one clone.